pahfit/base.py

```diff
@@ -463,6 +469,19 @@ def save(self, obs_fit, filename, outform):
                     ]
                 )
             elif comp_type == "Drude1D":
+
+                # Calculate feature strength.
+                # TODO: Use input units to determine conversion and feature strength units.
+                strength = (np.pi * const.c.to('micron/s').value / 2) * (component.amplitude.value * component.fwhm.value / component.x_0.value**2) * 1e-26
+                # Estimate profile-weighted uncertainty.
+                if strength != 0.:
+                    nu = const.c.to('micron/s').value / x.value
+                    dnu = np.diff(nu, prepend=nu[1])
+                    uwv = np.where(component(x.value) / np.nanmax(component(x.value)) > 0.1)[0]
+                    strength_unc = np.sqrt(np.sum(np.take(yerr, uwv)**2 * np.take(dnu, uwv)**2)) * 1e-26
```

Comment:
This is similar to estimate 1 in the IDL code: pahfit_main_feature_power.pro which calculates the strength of the main PAH features (combining multiple Drude profiles where necessary). This is different from the uncertainty calculated for each Drude profile in pahfit.pro which is given as the standard output for IDL pahfit. 
Should we be consistent with IDL pahfit and use the latter instead here?
Also, a bit more documentation about the method used would be helpful (e.g. copied from PAHFIT main feature power " profile-weighted quadrature summed sigma over the range of interest, W/m^2").

Reply:
I do suggest we leave all "residuals from the model" uncertainty estimates out of the main core code.  That was included as a bit of a hack in IDL PAHFIT, several years after it was first written, when I noticed what a poor job pipeline uncertainties (and hence PAHFIT-derived feature uncertainties) were doing at moderate to high SNR.  Since this will depend on the quality of the model, whether input spectra are sensible for PAHFIT'ing, instrumental details, etc., I think we should get pipeline uncertainties working cleanly with the full covariance matrix, and then perhaps add another convenience routine, maybe something like `alt_uncertainty`, for people who'd like alternative uncertainty estimates after the fitting.

Reply:
> This is similar to estimate 1 in the IDL code: pahfit_main_feature_power.pro which calculates the strength of the main PAH features (combining multiple Drude profiles where necessary). This is different from the uncertainty calculated for each Drude profile in pahfit.pro which is given as the standard output for IDL pahfit.
> Should we be consistent with IDL pahfit and use the latter instead here?
> Also, a bit more documentation about the method used would be helpful (e.g. copied from PAHFIT main feature power " profile-weighted quadrature summed sigma over the range of interest, W/m^2").

As we have extensively discussed yesterday, these are the only type of uncertainties that we can calculate at the moment given that the uncertainties from the model are not returned, and as JD mentioned above, this can become and alternative/optional uncertainty estimation in the future. So think of this as a temporary solution as well as an additional feature. This uncertainty calculation can be applied to the individual PAH components as well as the combined ones.

Reply:
Thanks. I can see that. As JD mentioned, should it be included in the main code for now or do we already split it off in another one?

Reply:
Can it be split off now, but then used by default?

Reply:
Until we have a better solution?

Reply:
> Can it be split off now, but then used by default?

Do you mean have the feature strength calculation as a separate module/file imported in base.py?

Reply:
Probably better to have it imported and called in run_pahfit.py.

Reply:
I could set it up as a separate module, and then run_pahfit can pass siimply an additional argument (e.g. strength_calc=True) to the save function, and the save function would then import and use the new module. Does it make sense?

Reply:
Yes.  By default, it should run the strength calc.